casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -53,34 +61,163 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
+      "!X (!4)"      -> merges(S0, S0 | F_, Nil),
+      "!X !C"        -> conflictsForNow(S0, S1, C1),
+      "!X (!C)"      -> conflicts(S0, S1 | C_, Nil),
+      "!X 4X"        -> conflicts(S0, F_, Nil),
+      "!X 4!"        -> conflictsForNow(S0, F_, S0),
+      "!X (4!)"      -> coveredBy("!X (!4)"),
+      "!X 4!!"       -> conflictsForNow(S0, F_, R0),
+      "!X (4!!)"     -> conflictsForNow(S0, F_ | R0, Nil),
+      "!X !!X"       -> conflictsForNow(S0, R0, Nil),
+      "!X !!4"       -> conflictsForNow(S0, R1, F1),
+      "!X (!!4)"     -> coveredBy("!X (4!!)"),
+      "!X CX"        -> conflicts(S0, C_, Nil),
+      "!X C!"        -> conflicts(S0, C_, S0),
+      "!X (C!)"      -> conflicts(S0, C_ | S0, Nil),
+      "!4 !4 same 4" -> conflicts(S0, S1, F_),
+      "!4 !4 diff 4" -> conflicts(S0, S1, F0 | F1),
+      "!4 (!4)"      -> merges(S0, S1 | F1, F0),
+      "(!4) (!4)"    -> merges(S0 | F_, S0 | F_, Nil),
+      "!4 !C"        -> conflictsForNow(S0, S1, F0 | C1),
+      "!4 4X"        -> conflictsForNow(S0, F_, F_),
+      "!4 4!"        -> conflictsForNow(S0, F_, F0 | S1),
+      "!4 4!!"       -> conflictsForNow(S0, F_, F0 | R1),
+      "!4 !!X"       -> conflictsForNow(S0, R1, F0),
+      "!4 !!4"       -> conflictsForNow(S0, R1, F_ | F1),
+      "!4 CX"        -> conflictsForNow(S0, C_, F_),
+      "!4 C!"        -> conflictsForNow(S0, C_, F0 | S1),
+      "!C !C"        -> conflictsForNow(S0, S0, C_),
+      "!C 4X"        -> conflictsForNow(S0, F_, C_),
+      "!C 4!"        -> conflictsForNow(S0, F_, C0 | S1),
+      "!C 4!!"       -> conflictsForNow(S0, F_, C0 | R1),
+      "!C !!X"       -> conflictsForNow(S0, R1, C0),
+      "!C !!4"       -> conflictsForNow(S0, R1, C0 | F1),
+      "!C CX"        -> conflictsForNow(S0, C_, C_),
+      "!C C!"        -> conflictsForNow(S0, C_, C0 | S1),
+      "4X 4X"        -> conflictsForNow(F_, F_, Nil),
+      "4X 4!"        -> conflicts(F_, F_, S0),
+      // Skipping 4X 4!! merges, 4X !!X may merge or not, 4X !!4 may merge or not
+      "4X CX"        -> conflicts(F_, C_, Nil),
+      "4X C!"        -> conflictsForNow(F0, C1, S1),
+      "4X (!!4)"     -> conflicts(F_, R0 | F_, Nil),
+      "4! 4! same !" -> conflicts(F_, F_, S0),
+      "4! 4! diff !" -> conflictsForNow(F0, F1, S0 | S1),
+      // Skipping 4! 4!! merges, 4! !!X merges, 4! !!4 merges
+      "4! CX"        -> conflictsForNow(F_, C1, S0),
+      "4! C! same !" -> conflictsForNow(F_, C_, S0),
+      "4! C! diff !" -> conflictsForNow(F0, C1, S0 | S1),
+      "CX CX"        -> conflictsForNow(C_, C_, Nil),
+      "C! C! same !" -> conflicts(C_, C_, S0),
+      "C! C! diff !" -> conflictsForNow(C0, C1, S0 | S1),
+      // 4!! / !!4 row is similar to !4 / 4! and thus skipped
+      // C!! / !!C row is similar to !C / C! and thus skipped
+      "CX !!X no match" -> conflictsForNow(R0, C1, Nil),
+      "CX !!X match"    -> conflicts(R0, C0, Nil)
+    ).values.toList.parSequence_
```

Comment:
We should verify that we have cases generated by the following code,
which amounts to 69 cases (not including same/diff distinction).

We'd check that by:

having the Map defining the test cases named `tests`
asserting that all cases `c` defined in `withLocals` have a key in the `tests` map that starts with `c`

```

val all = List("!X", "!4", "!C", "4X", "4!", "4!!", "PX", "P!", "P!!", "!!X", "!!4", "!!C", "CX", "C!", "C!!")
val considered = all.filterNot(_.contains("P")).filterNot(_.contains("!!"))

val pairs = considered.combinations(2)
val diagonal = considered.map(x => List(x, x))
val cases = (pairs ++ diagonal).toList

def isComm(s: String) = !s.contains("X")
def makeLocal(s: String): List[String] = isComm(s) match {
  case false => List(s)
  case true => List(s, s"($s)")
}

def makeLocals(v: List[String]): List[List[String]] = for {
  a <- makeLocal(v(0))
  b <- makeLocal(v(1))
} yield List(a, b)

val withLocals = cases.flatMap(makeLocals)

val s = withLocals.map(_.mkString(" ")).mkString("\n")
```

output:

```
!X !4
!X (!4)
!X !C
!X (!C)
!X 4X
!X 4!
!X (4!)
!X CX
!X C!
!X (C!)
!4 !C
!4 (!C)
(!4) !C
(!4) (!C)
!4 4X
(!4) 4X
!4 4!
!4 (4!)
(!4) 4!
(!4) (4!)
!4 CX
(!4) CX
!4 C!
!4 (C!)
(!4) C!
(!4) (C!)
!C 4X
(!C) 4X
!C 4!
!C (4!)
(!C) 4!
(!C) (4!)
!C CX
(!C) CX
!C C!
!C (C!)
(!C) C!
(!C) (C!)
4X 4!
4X (4!)
4X CX
4X C!
4X (C!)
4! CX
(4!) CX
4! C!
4! (C!)
(4!) C!
(4!) (C!)
CX C!
CX (C!)
!X !X
!4 !4
!4 (!4)
(!4) !4
(!4) (!4)
!C !C
!C (!C)
(!C) !C
(!C) (!C)
4X 4X
4! 4!
4! (4!)
(4!) 4!
(4!) (4!)
CX CX
C! C!
C! (C!)
(C!) C!
(C!) (C!)
```

Reply:
Ok, in a follow up PR